cargo-public-api/src/main.rs

```diff
@@ -130,11 +130,48 @@ fn main_() -> Result<()> {
 fn check_diff(args: &Args, diff: &Option<PublicItemsDiff>) -> Result<()> {
     match (&args.deny, diff) {
         // We were requested to deny diffs, so make sure there is no diff
-        (Some(_deny), Some(diff)) => {
-            if diff.is_empty() {
+        (Some(deny), Some(diff)) => {
+            let mut violations = crate::error::Violations::new();
+            for d in deny {
+                match d {
+                    DenyMethod::All => {
+                        if !diff.added.is_empty() {
+                            violations.extend_added(diff.added.iter().cloned());
+                        }
+                        if !diff.changed.is_empty() {
+                            violations.extend_changed(diff.changed.iter().cloned());
+                        }
+                        if !diff.removed.is_empty() {
+                            violations.extend_removed(diff.removed.iter().cloned());
+                        }
+                    }
+
+                    DenyMethod::Added => {
+                        if !diff.added.is_empty() {
+                            violations.extend_added(diff.added.iter().cloned());
+                        }
+                    }
+
+                    DenyMethod::Changed => {
+                        if !diff.changed.is_empty() {
+                            violations.extend_changed(diff.changed.iter().cloned());
+                        }
+                    }
+
+                    DenyMethod::Removed => {
+                        if !diff.removed.is_empty() {
+                            violations.extend_removed(diff.removed.iter().cloned());
+                        }
+                    }
+                }
```

Comment:
Here is how I would simplify this:

```suggestion
                let deny_added = matches!(d, DenyMethod::Added | DenyMethod::All);
                let deny_changed = matches!(d, DenyMethod::Changed | DenyMethod::All);
                let deny_removed = matches!(d, DenyMethod::Removed | DenyMethod::All);
                if deny_added && !diff.added.is_empty() {
                    violations.extend_added(diff.added.iter().cloned());
                }
                if deny_changed && !diff.changed.is_empty() {
                    violations.extend_changed(diff.changed.iter().cloned());
                }
                if deny_removed && !diff.removed.is_empty() {
                    violations.extend_removed(diff.removed.iter().cloned());
                }
```

Reply:
We could even further simplify this by adding functions on `DenyMethod` for indicating... 

```rust
pub fn is_deny_added(&self) -> bool {
    std::matches(self, DenyMethod::Added | DenyMethod::All)
}
```

and use them.

Reply:
Sounds great!